pkgs/native_assets_cli/lib/src/api/build_state.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'asset.dart';
+import 'build_config.dart';
+import 'build_output.dart';
+
+final class BuildState {
+  /// Run a native assets build.
+  ///
+  /// Example using `package:native_toolchain_c`:
+  ///
+  /// ```dart
+  /// void main(List<String> args) async =>
+  ///     await BuildState.build(args, (buildState) async {
+  ///       final cbuilder = CBuilder.library(
+  ///         name: packageName,
+  ///         assetId: 'package:$packageName/${packageName}.dart',
+  ///         sources: [
+  ///           'src/$packageName.c',
+  ///         ],
+  ///       );
+  ///       await cbuilder.run(
+  ///         buildState: buildState,
```

Comment:
Does `cbuilder.run` already take the brand new `BuildState` as argument?
Or is that a separate change that also needs to be done?

Reply:
This needs to be done later. Published dependencies means this must be landed and published before CBuilder can be updated.